library/java/net/openid/appauth/browser/BrowserSelector.java

```diff
@@ -91,15 +97,29 @@
             }
 
             try {
+                int defaultBrowserIndex = 0;
                 PackageInfo packageInfo = pm.getPackageInfo(
                         info.activityInfo.packageName,
                         PackageManager.GET_SIGNATURES);
 
                 if (hasWarmupService(pm, info.activityInfo.packageName)) {
-                    browsers.add(new BrowserDescriptor(packageInfo, true));
+                    BrowserDescriptor customTabBrowserDescriptor =
```

Comment:
Could you add tests to BrowserSelectorTest that demonstrate:

- If the user has no default browser, the order of the returned browsers matches the order of matching intents returned by `queryIntentActivities`. This case is likely already covered by the existing tests, but it will help to make that more explicit.
- If the user has a default browser, with custom tab support, the two entries for that browser are at the start of the list, with the custom tab descriptor first.
- If the user has a default browser, without custom tab support, that one entry for that browser is at the start of the list.

The code looks correct to me, but I just want to make sure that the changes are appropriately covered by tests so that future changes don't result in regressions.

Reply:
I am trying to look into creating a useful test.
Looks like I will be able to set the default browser, if I will use ShadowPackageManager, where I am able to use methods that are normally only possible to use by the system.
Will work on it tomorrow.

Reply:
The tests have now been added.